appcelerator/titanium_mobile
# fix(android): Fixed bug where Label "ellipsize" modes START/MIDDLE don't work if "autoLink" is true

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-24881

**Summary:**
- This was a limitation in Google's `LinkMovementMethod` Java class which only supported an `END` ellipsize mode.
- Google's `TextView` Java class on Android 4.4 and above supports links without a `MovementMethod`. So, the solution is to remove its usage since Titanium 8.0.0 min Android version supported is 4.4.

**Test:**
_(The below test **MUST** be ran on Android 4.4 as well.)_
1. Build and run the below code on Android.
2. Verify that ellipsis `...` are shown at the beginning of the text link.
3. Tap on the link and verify that it takes you to: https://www.appcelerator.com
4. Comment out the below code's `Ti.UI.TEXT_ELLIPSIZE_TRUNCATE_START` line.
5. Uncomment the below code's `Ti.UI.TEXT_ELLIPSIZE_TRUNCATE_MIDDLE` line.
6. Build and run on Android.
7. Verify that ellipsis `...` appear in the middle of the text link.
8. Tap on the link and verify that it takes you to: https://www.appcelerator.com
9. Comment out the below code's `Ti.UI.TEXT_ELLIPSIZE_TRUNCATE_MIDDLE` line.
10. Uncomment the below code's `Ti.UI.TEXT_ELLIPSIZE_TRUNCATE_END` line.
11. Build and run on Android.
12. verify that ellipsis `...` appear at the end of the text link.
13. Tap on the link and verify that it takes you to: https://www.appcelerator.com

```javascript
var window = Ti.UI.createWindow();
window.add(Ti.UI.createLabel({
	text: "https://www.appcelerator.com",
	font: { fontSize: "50dp" },
	autoLink: Ti.UI.AUTOLINK_URLS,
	ellipsize: Ti.UI.TEXT_ELLIPSIZE_TRUNCATE_START,
//	ellipsize: Ti.UI.TEXT_ELLIPSIZE_TRUNCATE_MIDDLE,
//	ellipsize: Ti.UI.TEXT_ELLIPSIZE_TRUNCATE_END,
	maxLines: 1,
	width: "100%",
}));
window.open();
```
